Allow multiple passes in cvs files, add function to produce minutes, allow empty comments

ariffero opened the pull request:
Changes:
1. I have modified the structure of the csv file in a way that it is possible to have all the runs of a certain period, for the different passes, in the same csv file. It is possible to choose which runs flag in the official RCTs when filling the csv file, and it is possible to choose the pass when running the script

2. I have added two option that allow to produce the minutes for the aQC meeting in case there is nothing strange. At the moment the only flags considered for the minutes are: good, bad tracking, and limited acceptance (MC reproducible), that are the ones we use for the MID.

3. I have added a workaround to use "empty" comments when running in batch mode. Running the script with empty comments made me the script to crash and so I added a check and when the comment in empty I replace it with a space.

Commit:
Add gitignore

Commit:
Add option to create minutes

Use the option --minutes file.txt to create a file called file.txt containing the minutes for the aQC meeting. The minutes are created basing on the cvs file passed in input.
This option is available only using the batch mode.

ariffero committed:
Merge branch 'JianLIUhep:main' into main

Commit:
Allow multiple passes in the .csv file and choose runs to post.

Commit:
Add workaround to allow empty comments in batch mode.

If the comment was empty the script crashed, the workaround is to replace the empty comment with a space. The replacement with a string "" does the script not to crash, but the runs are not posted on the RCTs.

Commit:
Bug fix: the workaround for empty comments wasn't working properly

JianLIUhep commented on README.md:
````diff
@@ -42,11 +42,14 @@ Currently, only run-based (covering the whole run duration) flags can be added w
+```
+python3 rct_post_flag.py rct_post_flag.json --data_pass "LHC24al_cpass0" --detector "ITS" -b test.csv --minutes minutes.txt --no_diff
 ```
-The format of the file refers to the Example.csv.
````
Can you please remove Example.csv from the repository?

ariffero replied:
Sure I have just committed this change.

JianLIUhep commented on README.md:
````diff
@@ -42,11 +42,14 @@ Currently, only run-based (covering the whole run duration) flags can be added w
-python3 test_rct_post_flag.py test_rct_post_flag.json --data_pass "LHC24al_cpass0" --detector "ITS" -b ITS_flag.csv
+python3 rct_post_flag.py rct_post_flag.json --data_pass "LHC24al_cpass0" --detector "ITS" -b test.csv
+```
+- It is possible to have in output a file that contains the minutes for the aQC meeting, based on the posted runs. To produce it use the argument `--minutes <name_of_the_file.txt>`, if the file is not present it will be generated and filled, otherwise the new info will be added to that file. If you are checking a run in a pass that is not `apass0` you may want to say explicitly that the quality o ...
````
I suggest using either `cpass0` or `apass1`.

ariffero replied:
I have change `apass0` to `cpass0`.

JianLIUhep reviewed (changes requested):
Thanks a lot for those very useful improvements. I just have a few minor comments. Then, let's merge it.

Commit:
Remove Example.csv

Commit:
Update README.md

JianLIUhep commented:
That would be great if you could add more flags. I would propose we add:  `Limited Acceptance (MC Not Reproducible)`, `Bad PID`, `No Detector Data` and `Unknown`. 

Thank you, Andrea!

Commit:
Add new flags in the minutes.

Added flags: Limited Acceptance (MC Not Reproducible), Bad PID, No Detector Data and Unknown.

ariffero commented:
I have added the flags you proposed, updated the README and added an example of the minutes produced by the script.

JianLIUhep commented:
Many thanks, @ariffero. Merging.
